### ISSUE TITLE
"Fatal error: Call to undefined fuction get_current_screen()" when accessing WordPress "Customize" admin screen

### ISSUE THREAD

**Issue by TechForJustice** [meitar/better-angels#119]
While trying to customize my website (after having already downloaded the Buoy Plugin) I ran into this error: 
"Fatal error: Call to undefined function get_current_screen() in […/path/to…]/wp-content/plugins/buoy/class-buoy-team.php on line 924"

**Comment by meitar** [comment]
It's possible you're still running an old version; uninstall and re-install the Buoy plugin and try again.

If it still happens, you'll need to provide more information than "While trying to customize my website" because that could mean anything.

Read http://www.noverse.com/blog/2012/06/how-to-write-a-good-bug-report/ for guidance on what to write in a bug report.

**Comment by vincentshadow** [comment]
I got a 500 too (Internal Server Error) upon activation of the Bouy plugin using the WordPress template theme AccessLite. When attemptinh to use the themes template customization feature (Customize) which is the themes dashboard essentially it is unable to load resulting in a "Failed to load resource: the server responded with a status of 500 (Internal Server Error)." 

When Bouy is deactivated then functionality returns. Have not attempted other template themes to recreate error. I am using the latest plugin. TechForJustice, what theme are you using?

**Comment by meitar** [comment]
Hm, thanks for the additional info. I'll reopen this and have a look ASAP.

**Comment by hannahhubbard** [comment]
TechForJustice is using WordPress theme SKT White Pro, and the latest plugin was installed. I just went and selected 'customize' on the themes dashboard to double check, and the error is the same - Fatal error: Call to undefined function get_current_screen() in […/path/to…]/wp-content/plugins/buoy/class-buoy-team.php on line 924.

**Comment by vincentshadow** [comment]
Thanks meitar. When I am back to that site I will look around and see what I can find too. Thx Hannah for confirming multiple themes.

**Comment by meitar** [comment]
Okay, two things. First, let's make sure we're on the same page about what's going on.

When you say:

> When attemptinh to use the themes template customization feature (Customize) which is the themes dashboard

I am assuming that you mean that you have clicked on Appearance -> Customize in your WordPress dashboard, resulting in the `wp-admin/customize.php` script being run. Yes?

If so, can either of you provide a link to the theme you are using? [I cannot find "AccessLite" on the WordPress plugin repository](https://wordpress.org/themes/accesslite/), and ["SKT White Pro" seems to be a for-pay theme](http://www.sktthemes.net/shop/skt-white-pro/), so I do not have its source code. This means I cannot confirm the bug, because I cannot reproduce it.

**Comment by meitar** [comment]
I found a theme "SKT White" (not the pro) and cannot reproduce the issue. I also found a theme named "AccessPress Lite" and thought maybe you had just mistyped the theme name, so I loaded it and tried to access the Customize screen there. It worked, there, too. :\

**Comment by hannahhubbard** [comment]
Yes, it's for-pay. If you're going to be up and want to investigate asap, I can give u admin access so you can check out the code. And, yes, we are on the same page - On the Dashboard, I click on Appearance>  Customize. Also discovered that it happens with Appearance > Header too.

**Comment by vincentshadow** [comment]
**Yes** and **yes**. I did mistype the theme, but you got it right (Accesspress Lite at https://wordpress.org/themes/accesspress-lite/

And yes, running the /wp-admin/customize.php was the issue.

I am running WP version 4.4.1 with no updates showing for themes, plugins... Can you confirm the latest version of bouy? 
[screenshot.pdf](https://github.com/meitar/better-angels/files/104492/screenshot.pdf)

**Comment by meitar** [comment]
> Also discovered that it happens with Appearance > Header too. 

"Header" is part of "Customize," so it makes sense that it would happen there, too. WordPress calls this the "[Theme Customizer](https://codex.wordpress.org/Appearance_Customize_Screen)."

> If you're going to be up and want to investigate asap, I can give u admin access so you can check out the code.

That would be handy, thanks. Send me a private message somehow (encrypted email, LastPass, something) to give me access credentials.

**Comment by meitar** [comment]
> Can you confirm the latest version of bouy? 

I haven't been strict about versions so, to date, the only way to "update to the latest version of Buoy" without touching a command line is to uninstall it from the WordPress interface (not just disable but totally uninstall it) and then re-install it anew. Currently, this will give you the tip of [`trunk/` stored in the WordPress plugin repository](https://plugins.trac.wordpress.org/browser/buoy/trunk).

Now that I know someone's actually installing it on their own site, I'll be stricter about updating the version string so we can communicate more sanely about what version of Buoy is running on a given site.

FWIW, I tested [Buoy at SVN revision 1328936](https://plugins.trac.wordpress.org/log/buoy/trunk/?rev=1328936) with the "AccessPress Lite" theme installed and active on my local site, and did not encounter the error. This either means you are not running the same Buoy code as I believe to be in that revision or that there is something else different about your install than mine. For instance, what PHP version are you running?

Any and all additional information would be helpful (and, really, necessary) at this point to figure out why you're seeing an error I can't reproduce.

**Comment by meitar** [comment]
> For instance, what PHP version are you running?

Your server reports Apache 2.x running PHP version 5.3.29. You also have at least one plugin that I can detect passively: W3 Total Cache. What happens when you deactivate this plugin? Does the problem persist?

I'll try installing that plugin on my test site to more closely mirror your environment but, again, it would be helpful if you could provide as complete a report as possible about your setup. There is only so much recon I can legally do against your site from over here…. ;)

**Comment by vincentshadow** [comment]
Reviewing in the Chrome developer console I am seeing the attached. Everything keeps pointing to load-scripts.php. Correct, I have PHP 5.3 and MySQL Version 5.5.44 on Apache baby. Tell me what else you need recon man. And ** everything is fine upon deactivation**.
[screenshot2.pdf](https://github.com/meitar/better-angels/files/104515/screenshot2.pdf)

**Comment by meitar** [comment]
> And everything is fine upon deactivation.

Deactivation of what? Buoy, or the other plugins?

**Comment by vincentshadow** [comment]
Activation of just Buoy = wp-admin/customize.php not loading and returning an error
Deactivation of just Buoy = wp-admin/customize.php loading and fine with no error

**Comment by vincentshadow** [comment]
I will re-install again in a bit and let you know what happens. afk, back in 30.

**Comment by meitar** [comment]
I've committed patch d30421a91539e4c5e2c5d4436c303d4f531aa876 that *seems* to resolve this issue for the TechForJustice.org site. Before I officially release it, though, @vincentshadow, can you please download and [manually replace your installed Buoy](https://codex.wordpress.org/Managing_Plugins#Manual_Plugin_Installation) with the version in [this zip file](https://github.com/meitar/better-angels/archive/5e336ab2fc5da7a096e3f12bbf7b2955db50b660.zip)?

Thanks.

**Comment by meitar** [comment]
For what it's worth, this is probably not an issue related to the themes. When I installed the SK White Pro theme on my local server, there was *still* no problem. So the problem is either some kind of weird plugin incompatibility or another combination of things that I haven't tracked down (and that may not be worth tracking down, given the simple workaround).

The code path for this is also strange to me. If you look at [WordPress 4.4.1's `customize.php` on line 13](https://core.trac.wordpress.org/browser/tags/4.4.1/src/wp-admin/customize.php#L13) you'll see a call to `require_once` the admin bootstrap. Tracing the execution path in [that file down to line 82](https://core.trac.wordpress.org/browser/tags/4.4.1/src/wp-admin/admin.php#L82) shows another load to the rest of the admin codebase, in `wp-admin/includes/admin.php`. And that file, in turn, [unconditionally loads the Screen API code (in lines 53 and 54)](https://core.trac.wordpress.org/browser/tags/4.4.1/src/wp-admin/includes/admin.php#L52), which is [where the `get_current_screen()` function is defined](https://core.trac.wordpress.org/browser/tags/4.4.1/src/wp-admin/includes/screen.php#L196), also unconditionally.

In other words, as far as I can tell, there is no obvious reason why the Screen API code would not be loaded for your WordPress and yet loaded without an issue in mine, and either way, it doesn't seem to be something that the theme you're using is causing.

So. That's weird.

**Comment by meitar** [comment]
Ping, @vincentshadow. Any news?

**Comment by meitar** [comment]
Okay so, I'm gonna assume my above patch fixes this because I haven't heard back from anyone about this in a week, and would like to release the patch sooner rather than later.

**Comment by hannahhubbard** [comment]
wait a min - know @vincentshadow touched on it yesterday - got three jobs and babies going on. will get back. sorry for delay.

**Comment by meitar** [comment]
I still haven't heard back from anyone about this and now it's approaching two weeks, so I'm not going to wait any longer, despite objections.
